tests/integration/test_hot_reload_storage_policy/test.py

```diff
@@ -74,13 +80,48 @@ def test_hot_reload_policy(started_cluster):
     node1.query(
         "CREATE TABLE t (d Int32, s String) ENGINE = ReplicatedMergeTree('/clickhouse/tables/t_mirror', '1') PARTITION BY d ORDER BY tuple() SETTINGS storage_policy = 'default_policy'"
     )
+    node1.query(
+        "CREATE TABLE t_d (d Int32, s String) ENGINE = Distributed('default', 'default', 't', d%20, 'default_policy')"
+    )
     set_config(node1, new_disk_config)
     time.sleep(1)
+
+    # After reloading new policy with new disk, merge tree tables should reinitialize the new disk (create relative path, 'detached' folder...)
+    # Otherwise FETCH PARTITION will fails
     node1.query("ALTER TABLE t FETCH PARTITION 1 FROM '/clickhouse/tables/t'")
+    node1.query("ALTER TABLE t ATTACH PARTITION 1")
+
+    # Check that fetch partition success and we get full data from node0
     result = int(node1.query("SELECT count() FROM t"))
     assert (
-        result == 4,
-        "Node should have 2 x full data (4 rows) after reloading storage configuration and fetch new partition, but get {} rows".format(
-            result
-        ),
+        result == 2
+    ), "Node should have 2 rows after reloading storage configuration and fetch new partition, but get {} rows".format(
```

Comment:
BTW you have chosen really sophisticated way to check disks updates, isn't the following is not the same?

>simply INSERT into MergeTree, then add new disks do few more INSERTs and check that all disks contains some parts?

_(since ReplicatedMergeTree requires ZooKeeper - 3 more nodes, obviously tests without ZooKeeper consume less resources, even thought this is minor)._

Reply:
>In this case, if detached folder is not created then the test will still pass. I want a test that will both write to disk and touch detached folder.

You can add few ATTACH/DETACH queries

>I took it from one of our CH usage

But sometimes tests that replicates the production adds extra guarantee that everything will work... :)

Reply:
Ok I'm convinced. I will make the test more simple.